### ISSUE TITLE
Movie desyncs; inaccurate vs real hardware

### ISSUE THREAD

**Issue by BillyWM** [SourMesen/Mesen#597]
This Legend of Zelda playthrough desyncs in the first dungeon, with Link dying in the corner.

However, this run has been console verified, so it must be attributable to emulation inaccuracy, be it timing or RAM initialization state etc

TAS: http://tasvideos.org/2508M.html
Console verification: https://www.youtube.com/watch?v=uNTR7vPkoqo

-------------------

Things I checked:

* That I'm using the right ROM. I am using the "PRG0" ROM with a checksum verified to match the checksum recorded in the movie file

* That revision doesn't matter: The video is verifying against the later "Rev A" cart with the rest/battery warning, while the movie was created with the earliest release, "PRG0" without this warning. I ran the movie in FCEUX using the RevA ROM like in the video and it makes no difference.

* Mesen versions: I checked this against both the stable version of Mesen and the latest dev build, doing a clean install with each.

**Comment by SourMesen** [comment]
Thanks for the report.  It's possible that some sort of inaccuracy could be at play, unfortunately trying to fix emulation accuracy through TAS movie comparisons is rather hard/time consuming.

Also, you forgot to mention an important detail: how did you play that movie in Mesen?  If you relied on Mesen's (hidden) ability to play FM2 files, it may not be very reliable (and I haven't tested it in well over a year)

**Comment by BillyWM** [comment]
Yes, I just opened it from the movie menu loading that .fm2 file. Is this something it does a lot when coaxing it into playing an FM2 movie?

**Comment by SourMesen** [comment]
A lot (most?) of FM2 movies desync, as far as I am aware of - the feature isn't advertised because it's not really useful in its current state.  Some BK2 movies keep in sync, but some desync, too.  There are so many variables to NES emulation (including different CPU/PPU alignments at power on that no emulator actually emulates) that it's pretty hard to realistically make movies sync between multiple emulators and hardware, too.

**Comment by SourMesen** [comment]
Rev A is definitely required to run this movie.  The game desyncs around frame 40k on the first version of the game (no rev), most likely due to DMC DMA screwing up the controls (this happens on FCEUX, too)

The desync at the start of the movie was simply because the emulator didn't ignore the save data on disk when playing the movie, which caused a desync in the menu at the start.  Fixing that and using rev A lets it sync to around frame ~45k, and which point it desyncs from FCEUX.  Unsure why, and unfortunately rather hard to debug something like this, so this may be as far as I can go here.
